### ISSUE TITLE
[Bug] KDA Triton backend remaps padding slot -1 to the last mamba pool slot, which is allocatable — padded rows can read and write a live request's SSM state (GLM-5.3-Flash, PR #36507)

### ISSUE THREAD

**Issue by ehsan** [sgl-project/sglang#36880]
Found while auditing GLM-5.3-Flash under DP attention (companion report:
https://github.com/sgl-project/sglang/issues/36879); this one is independent of DP and unfixed at PR #36507
head `aa8c950a3d`. We have not observed it in production — our configuration
does not produce the padded rows that trigger it — but the aliasing is real and
upstream should know before the model support merges.

## The bug

`TritonKDAKernel.extend`
(`python/sglang/srt/layers/attention/linear/kernels/kda_triton.py:239` at PR
#36507 head `aa8c950a3d`) routes padding rows away from the −1 sentinel before calling the
chunked kernel:

```python
# CP/DP padding marks zero-length requests with -1. The Triton state kernel
# performs raw pointer arithmetic on these indices, so route padding to
# MambaPool's trailing sentinel slot ...
ssm_cache_indices = torch.where(
    cache_indices >= 0, cache_indices, ssm_states.shape[0] - 1
)
```

The comment assumes the trailing slot is a reserved sentinel. It is not — the
reservation is at the **other end** of the pool:

- `MambaPool` allocates `size + 1` rows
  (`python/sglang/srt/mem_cache/memory_pool.py`).
- `MambaSlotAllocator` reserves **slot 0** as the dummy write target and hands
  out `1..size` inclusive
  (`python/sglang/srt/mem_cache/allocator/mamba.py:94` at `aa8c950a3d`:
  `# Slot 0 is reserved as a dummy write target for padded tokens.`
  `self.free_slots = torch.arange(1, self.size + 1, ...)`).

So `ssm_states.shape[0] - 1 == size`, the **last allocatable slot**. Any padded
row remapped there reads that slot's contents as its initial state and — via
`chunk_gated_delta_rule_fwd_h`'s `INPLACE_UPDATE` epilogue, whose `valid_state`
guard (`index >= 0`) is defeated by the remap — writes its final state back
into it. If a live request (or a mamba radix-tree checkpoint) currently owns
slot `size`, its recurrent state is silently corrupted; the request produces
wrong output with no error.

Note the sentinel-skip guard exists in the decode path
(`fused_recurrent`-family kernels skip `index < 0` rows) and the base GDN
chunked path got the equivalent fix in PR #33810; the KDA remap sidesteps both
by making the index non-negative before the kernel ever sees it.

## When it is reachable

Any batch where the KDA extend path receives rows with `mamba_cache_indices ==
−1`: DP/CP padding of extend batches, and the padded-row plumbing in
`hybrid_linear_attn_backend.py` (`mamba_cache_indices[_real_bs:] = -1`). The
slot is handed out under memory pressure (the free list reaches `size` once
enough concurrent requests + radix checkpoints are live), so the corruption
profile is "rare, load-dependent, per-request wrong output" — the worst kind to
debug from the outside.

## Suggested fix

Remap to slot **0**, matching the allocator's actual contract:

```python
ssm_cache_indices = torch.where(cache_indices >= 0, cache_indices, 0)
```

or teach the chunked kernel to honor the −1 sentinel the way the decode kernels
and post-#33810 GDN extend kernel do, and drop the remap entirely. Either way,
a comment on `MambaSlotAllocator` stating which slot is reserved would prevent
the two ends from disagreeing again.

## Related observations in the same path (minor, from the same audit)

- `chunk_kda` writes its output into the **caller's `v` tensor** in place
  (`chunk_kda_fwd` passes `o=v` to `chunk_gla_fwd_o_gk`). Callers today are
  saved by an incidental `.contiguous()` copy after `split()`; any future
  caller passing an already-contiguous `v` gets its input clobbered (we hit
  this in a standalone harness).
- The `T=1` chunked-KDA path reads unwritten memory: with
  `torch.use_deterministic_algorithms(True)` +
  `torch.utils.deterministic.fill_uninitialized_memory = True`, sequential
  single-token `chunk_kda` calls produce NaNs that propagate into the recurrent
  state. Benign inputs normally hide this (the region usually holds zeros), but
  it indicates a load without a proper bounds/init guard somewhere in the
  chunked pipeline at `T < chunk_size` edge shapes.

**Comment by ehsan** [comment]
Fix proposed in #36885.
